beacon-chain/blockchain/process_block.go

```diff
@@ -98,10 +99,17 @@ func (s *Service) onBlock(ctx context.Context, signed block.SignedBeaconBlock, b
 		return err
 	}
 
+	preStateVersion, preStateHeader, err := getStateVersionAndPayload(preState)
+	if err != nil {
+		return err
+	}
 	postState, err := transition.ExecuteStateTransition(ctx, preState, signed)
 	if err != nil {
 		return err
 	}
```

Comment:
Move the stateVersionAndPayload closer to its usage.

```suggestion
	postState, err := transition.ExecuteStateTransition(ctx, preState, signed)
	if err != nil {
		return err
	}
	preStateVersion, preStateHeader, err := getStateVersionAndPayload(preState)
	if err != nil {
		return err
	}
```

Reply:
`preState` gets mutated after calling `ExecuteStateTransition` so we must call `getStateVersionAndPayload` before `ExecuteStateTransition`.  I'll add a comment

Reply:
OK. After discussing offline, I am unhappy about `transition.ExecuteStateTransition` mutating the argument state AND returning a post state, but it is out of scope for this PR.